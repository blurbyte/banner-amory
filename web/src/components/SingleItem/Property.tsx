// item propert with a label
// for example: name - alette's bracelet

import * as React from 'react';
import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '../../styles/colors';

const Row = styled.dl`
  margin: 1rem 0;
`;

const Label = styled.dt`
  font-size: 1.2rem;
  color: ${transparentize(0.5, Colors.blue)};
  text-transform: uppercase;
  letter-spacing: 1px;
`;

type DetailProps = {
  big?: boolean;
};

const Detail = styled.dd<DetailProps>`
  font-size: ${props => (props.big ? '1.8rem' : '1.4rem')};
  font-weight: ${props => (props.big ? 600 : 'inherit')};
  margin: 0;
`;

type PropertyProps = {
  children: React.ReactNode;
};

class Property extends React.Component<PropertyProps> {
  static Label = Label;

  static Detail = Detail;

  render() {
    const { children } = this.props;
    return <Row>{children}</Row>;
  }
}

export default Property;
