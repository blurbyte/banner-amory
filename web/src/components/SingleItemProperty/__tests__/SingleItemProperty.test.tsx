import * as React from 'react';
import { render, cleanup } from 'react-testing-library';

import Property from '../index';

beforeEach(cleanup);

test('renders correctly', () => {
  const { container } = render(
    <Property>
      <Property.Label>Taylor</Property.Label>
      <Property.Detail>Swift</Property.Detail>
    </Property>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('statLine is passed correctly to children', () => {
  const { container } = render(
    <Property statLine>
      <Property.Label>Taylor</Property.Label>
      <Property.Detail>Swift</Property.Detail>
    </Property>
  );
  expect(container.firstChild).toMatchSnapshot();
});
