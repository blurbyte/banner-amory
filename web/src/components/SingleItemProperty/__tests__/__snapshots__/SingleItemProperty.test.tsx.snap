// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<dl
  class="sc-bdVaJa icGVfW"
>
  <dt
    class="sc-bxivhb cTGsno sc-htpNat gPRtVk"
  >
    Taylor
  </dt>
  <dd
    class="sc-ifAKCX jjAkxp"
  >
    Swift
  </dd>
</dl>
`;

exports[`statLine is passed correctly to children 1`] = `
<dl
  class="sc-bdVaJa lfQhji"
>
  <dt
    class="sc-bxivhb laDCst sc-htpNat gPRtVk"
  >
    Taylor
  </dt>
  <dd
    class="sc-ifAKCX imMJrI"
  >
    Swift
  </dd>
</dl>
`;
